plonky2/src/plonk/proof.rs

```diff
@@ -98,12 +103,13 @@ impl<F: RichField + Extendable<D>, C: GenericConfig<D, F = F>, const D: usize>
         C::InnerHasher::hash_no_pad(&self.public_inputs)
     }
 
-    pub fn to_bytes(&self) -> anyhow::Result<Vec<u8>> {
-        let mut buffer = Buffer::new(Vec::new());
-        buffer.write_proof_with_public_inputs(self)?;
-        Ok(buffer.bytes())
+    pub fn to_bytes(&self) -> Vec<u8> {
+        let mut buffer = Vec::new();
+        let _ = buffer.write_proof_with_public_inputs(self);
```

Comment:
To make it more obvious that this is safe, I'd rather we call unwrap or expect.

Or since we only write to `Vec`s (directly or through a `Cursor`), maybe we should change `Write` to not return any errors? If so it could be a future change though.

Reply:
Technically writing to a `Cursor<Vec<u8>>` (using `std::io`) can error whereas writing to a `Vec` (using its `push` APIs) cannot. I'll add the `expect` to make it more clear, but in general we have a stronger guarantee here than a regular `expect`.